scripts/gen_app_partitions.py

```diff
@@ -149,6 +150,26 @@ def parse_obj_files(partitions):
                     find_obj_file_partitions(fullname, partitions)
 
 
+def parse_compile_command_file(partitions):
+    # Iterate over all entries to find object files.
+    # Thereafter process each object file to find partitions
+    object_pattern = re.compile(r'-o\s+(\S*)')
+    with open(args.compile_commands_file, 'rb') as f:
+        commands = json.load(f)
+        for command in commands:
+            build_dir = command.get('directory')
+            compile_command = command.get('command')
+            compile_arg = object_pattern.search(compile_command)
+            obj_file = None if compile_arg is None else compile_arg.group(1)
+            if obj_file is not None:
```

Comment:
IIRC

if obj_file:

is more pythonic

Reply:
fixed.